x-pack/plugins/security_solution/public/common/components/markdown_editor/editor.tsx

```diff
@@ -73,9 +74,16 @@ const MarkdownEditorComponent = forwardRef<MarkdownEditorRef, MarkdownEditorProp
 
     const licenseIsPlatinum = useLicense().isPlatinumPlus();
 
+    const upsellInvestigationGuide = useUpsellingComponent('investigation_guide');
```

Comment:
Maybe it would be helpful to separate the component and string upseliing hooks, having: `useUpsellingComponent = (id: string) => ComponentType | null` 
and 
`useUpsellingMessage = (id: string) => string | null` 
hooks. And do the _isString_ check inside. Otherwise, we'll have to do the !isString everywhere to make TS happy.